unidist/core/backends/mpi/core/communication.py

```diff
@@ -129,6 +130,26 @@ def __init__(self, comm):
             self.topology[host][host_rank] = global_rank
             self.host_by_rank[global_rank] = host
 
+        mpi_hosts = MpiHosts.get()
+        if mpi_hosts is not None:
+            host_list = mpi_hosts.split(",") if mpi_hosts is not None else ["localhost"]
+            host_count = len(host_list)
+
+            # check running hosts
+            if self.is_root_process() and len(self.topology.keys()) > host_count:
+                print(
+                    """WARNING: The number of running hosts is greater than that specified in the UNIDIST_MPI_HOSTS.
+                    If you want to run the program on a host other than the local one, specify the appropriate parameter for `mpiexec`
+                    (`--host` for OpenMPI or `--hosts` for other MPI implementations)
+                """
+                )
+
+            # check running hosts
+            if self.is_root_process() and len(self.topology.keys()) < host_count:
+                print(
```

Comment:
Should this be an exception?

Reply:
No, because all processes may not be placed on all hosts and work correctly, but we should warn the user about this.